src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -576,6 +576,12 @@ private object GetVSSolutionProperty(int propId)
         {
             Debug.Assert(ThreadHelper.CheckAccess());
 
+            // if VSSolutionManager Initialization fail, _vsSolution can be null.
+            if (_vsSolution == null)
```

Comment:
I would say move this null check in `InitializeAsync` since that's the place we get IVsSolution service and also use `Assumes.Present` to do null check since it will throw appropriate error message if service is null.